docs/api_docs/matlab.rst

```diff
@@ -1,66 +1,153 @@
+Opening a Serial Connection
+------------------------------
 
-``client_object.save(’short_name’);``
+In order to connect with your USB-to-UART bridge, the MATLAB API provides the ``MessageInterface`` class. The ``MessageInterface`` class is initialized with a serial 
+port name and a baud rate. The name of your serial port is dependent on your operating system. For example, in the hardware example above, the serial port name is 
+“COM3,” and in Linux, it may be "/dev/ttyUSB0."
 
-sends a save message, which store’s the current RAM value into non-volatile memory. These functions are
-blocking and perform all necessary tasks for messaging.
+Suppose your FTDI reports on COM4, and you would like to communicate with your module at 921600 baud. To do so, we can create a ``MessageInterface`` ``com`` by 
+running ``com = MessageInterface('COM4', 921600)``.
 
+The ``MessageInterface`` is responsible for creating, transmitting, receiving, and decoding IQUART messages. All further communication is handled by ``com``.
 
-All clients have added member functions list, get all, set all, set verify, and save all.
+Interacting with Clients and Endpoints
+-------------------------------------------
+As mentioned in :ref:`getting_started_with_apis`, all Vertiq clients contain endpoints that can accept get, set, and save commands. This section discusses how to perform gets, sets, and saves through the MATLAB API.
 
-``client_object.list()``
+Before moving forward, please familiarize yourself with the clients available for your module's family and firmware style. You can find this information on your 
+module's family page in the Supported IQUART Clients section.
 
-displays all possible short names, their data types, and their units.
+Each Vertiq client has its own MATLAB implementation. You can find a full listing of available `MATLAB clients in the API <https://github.com/iq-motion-control/iq-module-communication-matlab/tree/master/iq/src/clients>`_. 
+Each MATLAB client is linked with a json file stored in a `git submodule <https://github.com/iq-motion-control/client_files/tree/74284e59abbe86e2b6d629a3727163a28254fd70>`_ that specifies the entries within each client.
 
-``data_all = client_object.get_all()``
+In order to create an instance of a client object, you must have at least a ``MessageInterface`` (``com``), and optionally a target Module ID (``obj_id``) 
+and timeout (``timeout``). By default the MATLAB API assumes a Module ID of 0 and a timeout of 0.2.
 
-performs a get on all messages in list and stores it in data all.
+Suppose we wanted to create an instance of the Microcontroller Temperature Monitor that uses our ``com`` instance created above, 
+communicates with a module with Module ID 42, and we allow up to 0.5 seconds for responses. To do so, we'd create an instance 
+of ``TemperatureMonitorUcClient`` with ``uc_temp = TemperatureMonitorUcClient(‘com’, com, ‘obj_id’, 42, ‘timeout’, 0.5);``
```

Comment:
Also maybe good to break out to a little code block, I find people often just skim through for code blocks and pictures, so having important stuff pop out in blocks can be helpful